skopt/sample.py

```diff
@@ -0,0 +1,79 @@
+from collections import Mapping
+from numbers import Integral
+from numbers import Real
+
+from scipy.stats._distn_infrastructure import rv_frozen
+from scipy.stats.distributions import randint
+from scipy.stats.distributions import uniform
+
+from sklearn.utils import check_random_state
+from sklearn.utils.fixes import sp_version
+
+
+class Categorical:
+    def __init__(self, *args):
+        self.values = args
+
+    def rvs(self, random_state=None):
+        rng = check_random_state(random_state)
+        return self.values[rng.randint(len(self.values))]
+
+    @property
+    def low(self):
+        return min(self.values)
+
+    @property
+    def high(self):
+        return max(self.values)
+
+
+class Discrete:
+    def __init__(self, low, high):
+        self.low = low
+        self.high = high
+
+    def rvs(self, random_state=None):
+        return randint(self.low, self.high).rvs(random_state=random_state)
+
+
+class Continuous:
+    def __init__(self, low, high):
+        self.low = low
+        self.high = high
+
+    def rvs(self, random_state=None):
+        return uniform(self.low, self.high).rvs(random_state=random_state)
+
+
+def points(grid, random_state=None):
+    if isinstance(grid, Mapping):
+        grid = [grid]
+
+    for sub_grid in grid:
```

Comment:
When would you want to pass a `list of dict of objects?` (Not a rhetoric)

I wrongly thought that the idea was to provide some sort of prior. For example, `test_sub_grids` indicates that you want a higher chance of picking a number from 4-5 rather than from 1-4